packages/react-components/react-checkbox/src/components/Checkbox/useCheckboxStyles.ts

```diff
@@ -165,24 +177,24 @@ const useIndicatorBaseClassName = makeResetStyles({
   justifyContent: 'center',
   overflow: 'hidden',
 
-  border: tokens.strokeWidthThin + ' solid',
+  color: values.indicatorColor,
+  backgroundColor: values.indicatorBackgroundColor,
+  border: tokens.strokeWidthThin + ' solid ' + values.indicatorBorderColor,
   borderRadius: tokens.borderRadiusSmall,
   margin: tokens.spacingVerticalS + ' ' + tokens.spacingHorizontalS,
   fill: 'currentColor',
   pointerEvents: 'none',
 
-  fontSize: '12px',
-  height: indicatorSizeMedium,
-  width: indicatorSizeMedium,
-});
+  fontSize: `calc(${values.indicatorSize} - 4px)`,
+  height: values.indicatorSize,
+  width: values.indicatorSize,
 
-const useIndicatorStyles = makeStyles({
-  large: {
-    fontSize: '16px',
-    height: indicatorSizeLarge,
-    width: indicatorSizeLarge,
+  '> *': {
```

Comment:
This is an super expensive selector as it matches all elements on a page first.

Reply:
This `> *` selector already existed in the Checkbox styles; it's just being refactored in this PR:
```
[`:not(:checked):not(:indeterminate) ~ .${checkboxClassNames.indicator} > *`]
```

 I made a separate PR (unrelated to this one) to remove it, and could update this PR once it's in:
* #26248

Reply:
Removed